java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java

```diff
@@ -36,13 +36,13 @@
 @Rule(key = "S1313")
 public class HardcodedIpCheck extends BaseTreeVisitor implements JavaFileScanner {
 
-  private static final Pattern IP_V4_REGEX = Pattern.compile("([^\\d.]*\\/)?((?<ip>(?:\\d{1,3}\\.){3}\\d{1,3})(:\\d{1,5})?(?!\\d|\\.))(\\/.*)?");
+  private static final Pattern IP_V4_REGEX = Pattern.compile("([^\\d.]*\\/)?((?<ip>(?:\\d{1,3}\\.){3}\\d{1,3})(:\\d{1,5})?(?![\\d.]))(\\/.*)?");
```

Comment:
This will probably be more readable if we split the regex into multiple named (private static final String) variables, PROTOCOL, IP, PORT and PATH. That will hopefully make it more readable, fix the regex complexity issue and avoid the confusion I had initially about what the rule is even matching.

Reply:
The protocol, port and path parts could then also be reused in the IP_V6_URL pattern.